src/video_core/shader/shader_jit_a64_compiler.cpp

```diff
@@ -0,0 +1,1207 @@
+
+    FCMGE(SRC2.S4(), SRC1.S4(), SRC2.S4());
+    AND(SRC2.B16(), SRC2.B16(), ONE.B16());
+
+    Compile_DestEnable(instr, SRC2);
+}
+
+void JitShader::Compile_SLT(Instruction instr) {
+    if (instr.opcode.Value().EffectiveOpCode() == OpCode::Id::SLTI) {
+        Compile_SwizzleSrc(instr, 1, instr.common.src1i, SRC1);
+        Compile_SwizzleSrc(instr, 2, instr.common.src2i, SRC2);
+    } else {
+        Compile_SwizzleSrc(instr, 1, instr.common.src1, SRC1);
+        Compile_SwizzleSrc(instr, 2, instr.common.src2, SRC2);
+    }
+
+    FCMGT(SRC1.S4(), SRC2.S4(), SRC1.S4());
+    AND(SRC1.B16(), SRC1.B16(), ONE.B16());
+
+    Compile_DestEnable(instr, SRC1);
+}
+
+void JitShader::Compile_FLR(Instruction instr) {
+    Compile_SwizzleSrc(instr, 1, instr.common.src1, SRC1);
+    FRINTM(SRC1.S4(), SRC1.S4());
+    Compile_DestEnable(instr, SRC1);
+}
+
+void JitShader::Compile_MAX(Instruction instr) {
+    Compile_SwizzleSrc(instr, 1, instr.common.src1, SRC1);
+    Compile_SwizzleSrc(instr, 2, instr.common.src2, SRC2);
+
+    // VSCRATCH0 = Ordinal(SRC1)
+    FCMEQ(VSCRATCH0.S4(), SRC1.S4(), SRC1.S4());
+
+    // FMAX will always pick the NaN
+    FMAX(SRC1.S4(), SRC1.S4(), SRC2.S4());
+
+    // In the case of NaN, pick SRC2
+    BIF(SRC1.B16(), SRC2.B16(), VSCRATCH0.B16());
+
+    Compile_DestEnable(instr, SRC1);
+}
+
+void JitShader::Compile_MIN(Instruction instr) {
+    Compile_SwizzleSrc(instr, 1, instr.common.src1, SRC1);
+    Compile_SwizzleSrc(instr, 2, instr.common.src2, SRC2);
+
+    // VSCRATCH0 = Ordinal(SRC1)
+    FCMEQ(VSCRATCH0.S4(), SRC1.S4(), SRC1.S4());
+
+    // FMIN will always pick the NaN
+    FMIN(SRC1.S4(), SRC1.S4(), SRC2.S4());
+
+    // In the case of NaN, pick SRC2
+    BIF(SRC1.B16(), SRC2.B16(), VSCRATCH0.B16());
+
+    Compile_DestEnable(instr, SRC1);
+}
+
+void JitShader::Compile_MOVA(Instruction instr) {
+    SwizzlePattern swiz = {(*swizzle_data)[instr.common.operand_desc_id]};
+
+    if (!swiz.DestComponentEnabled(0) && !swiz.DestComponentEnabled(1)) {
+        return;
+    }
+
+    Compile_SwizzleSrc(instr, 1, instr.common.src1, SRC1);
+
+    // Convert floats to integers using truncation (only care about X and Y components)
+    FCVTZS(SRC1.S4(), SRC1.S4());
+
+    // Get result
+    MOV(XSCRATCH0, SRC1.Delem()[0]);
+
+    // Handle destination enable
+    if (swiz.DestComponentEnabled(0) && swiz.DestComponentEnabled(1)) {
+        // Move and sign-extend low 32 bits
+        SXTW(ADDROFFS_REG_0, XSCRATCH0.toW());
+
+        // Move and sign-extend high 32 bits
+        LSR(XSCRATCH0, XSCRATCH0, 32);
+        SXTW(ADDROFFS_REG_1, XSCRATCH0.toW());
+    } else {
+        if (swiz.DestComponentEnabled(0)) {
+            // Move and sign-extend low 32 bits
+            SXTW(ADDROFFS_REG_0, XSCRATCH0.toW());
+        } else if (swiz.DestComponentEnabled(1)) {
+            // Move and sign-extend high 32 bits
+            LSR(XSCRATCH0, XSCRATCH0, 32);
+            SXTW(ADDROFFS_REG_1, XSCRATCH0.toW());
+        }
```

Comment:
```suggestion
        if (swiz.DestComponentEnabled(0)) {
            // Move and sign-extend low 32 bits
            SXTW(ADDROFFS_REG_0, XSCRATCH0.toW());
        } 
        if (swiz.DestComponentEnabled(1)) {
            // Move and sign-extend high 32 bits
            LSR(XSCRATCH0, XSCRATCH0, 32);
            SXTW(ADDROFFS_REG_1, XSCRATCH0.toW());
        }
```
Could this be simplified like this?

Reply:
Also you can probably use `ASR` instead of LSR + SXTW